src/content/en/updates/2020/09/migrating-to-es-modules.md

```diff
@@ -0,0 +1,256 @@
+
+In the end, updating the very first folder to include `export`-statements [took about a week](https://chromium-review.googlesource.com/c/chromium/src/+/1816563) and [multiple attempts with relands](https://chromium-review.googlesource.com/c/chromium/src/+/1825420).
+
+### `import`-phase
+
+After all symbols are both exported using `export`-statements and remained on the global scope (legacy), we had to update all references to cross-file symbols to use ES imports.
+The end goal would be to remove all "legacy export objects”, cleaning up the global scope.
+The transformation would be automated, by [running a script per folder](https://chromium-review.googlesource.com/c/devtools/devtools-frontend/+/1958525).
+
+For example, for the following symbols that exist in the `module.json` world:
+
+```js
+Module.File1.exported();
+AnotherModule.AnotherFile.alsoExported();
+SameModule.AnotherFile.moduleScoped();
+```
+
+They would be transformed to:
+
+```js
+import * as Module from '../module/Module.js';
+import * as AnotherModule from '../another_module/AnotherModule.js';
+
+import {moduleScoped} from './AnotherFile.js';
+
+Module.File1.exported();
+AnotherModule.AnotherFile.alsoExported();
+moduleScoped();
+```
+
+However, there were some caveats with this approach:
+
+1. Not every symbol was named as `Module.File.symbolName`.
+Some symbols were named solely `Module.File` or even `Module.CompletelyDifferentName`.
+This inconsistency meant that we had to create an internal mapping from the old global object to the new imported object.
+2. Sometimes there would be clashes between moduleScoped names.
+Most prominently, we used a pattern of declaring certain types of `Events`, where each symbol was named just `Events`.
+This meant that if you were listening for multiple types of events declared in different files, a nameclash would occur on the `import`-statement for those `Events`.
+3. As it turned out, there were circular dependencies between files.
+This was fine in a global scope context, as the usage of the symbol was after all code was loaded.
+However, if you require an `import`, the circular dependency would be made explicit.
+This isn't a problem immediately, unless you have side-effect function calls in your global scope code, which DevTools also had.
+All in all, it required some surgery and refactoring to make the transformation safe.
+
+## A whole new ES world
+
+In February 2020, 6 months after the start in September 2019, [the last cleanups](https://chromium-review.googlesource.com/c/devtools/devtools-frontend/+/2055066) were performed in the `ui/` folder.
+This marked the unofficial end to the migration.
+After letting the dust settle down, we officially marked the migration as [finished on March 5th 2020](https://bugs.chromium.org/p/chromium/issues/detail?id=1006759#c410). 🎉
```

Comment:
Added a party popper emoji here, I think it deserve it! 😆 @TimvdLippe

Reply:
Aw yeah 💪